functions/postIssue.js

```diff
@@ -0,0 +1,59 @@
+const algoliasearch = require('algoliasearch')
+
+// A Netlify function to insert a new record to Algolia
+exports.handler = async event => {
+  // Parsing and validating request
+  if (event.httpMethod !== 'POST') {
+    return {
+      statusCode: 405,
+      body: 'Method Not Allowed',
+    }
+  }
+
+  const bodyJson = JSON.parse(event.body)
+  const { id, ...rest } = bodyJson
+
+  // TODO - validate body arguments
+
+  // Retrieving environment variables.
+  // These variable should be configured on Netlify.
+  const algoliaApiId = process.env.ALGOLIA_APP_ID
+  const algoliaApiKey = process.env.ALGOLIA_API_KEY
+  const algoliaIndexName = process.env.ALGOLIA_INDEX_NAME
+
+  if (!algoliaApiId || !algoliaApiKey || !algoliaIndexName) {
+    return {
+      statusCode: 500,
+      body: 'Failed to execute function. Some environment variables are missing',
+    }
+  }
+
+  // Connecting to the Algolia service
+  const client = algoliasearch(algoliaApiId, algoliaApiKey)
+  const index = client.initIndex(algoliaIndexName)
+
+  // Send the record to Algolia
+  try {
+    const result = index
+      .saveObject(
+        {
+          objectID: id,
+          ...rest,
+        },
+        {
+          autoGenerateObjectIDIfNotExist: true, // TODO - make sure ID is generated in our code
+        }
+      )
+      .wait()
+  } catch (error) {
+    return {
+      statusCode: 500,
+      body: 'Failed to execute function. Could not save record in the database',
+    }
+  }
+
+  return {
+    statusCode: 200,
+    body: 'Data was inserted posted successfully',
```

Comment:
Is there data in the `result` that we should return to the client?

Reply:
The client already has the `issues` object. The only thing it needs is a confirmation that the issue was posted. I'm using the `.wait()` method of Algolia's client to make sure the data was saved.